backend/src/passwords.cpp

```diff
@@ -0,0 +1,72 @@
+#include <iostream>
+#include "passwords.hpp"
+
+namespace passwords
+{
+    std::unordered_set<std::string> generatePasswords(const int &numPasswords, const int &maxChars)
+    {
+        int minChars = 10;
+
+        if (maxChars < minChars) {
+            std::cout << "Password must have a minimum length of 10 characters\n";
+            exit(0);
+        }
+
+        std::unordered_set<std::string> passwords;
+        std::string lowercase = "abcdefghijklmnopqrstuvwxyz";
+        std::string uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        std::string digits = "0123456789";
+        std::string symbols = "!#$%&'()\\*+,-./:;<=>?@[]^_`{|}~\"";
+
+        while (passwords.size() < numPasswords)
+        {
+            // randomize length between minChars and maxChars
+            int length = minChars + rand() % ((maxChars + 1) - minChars);
+            std::string password = ""; 
+            bool hasLowercase = false, hasUppercase = false, hasDigit = false, hasSymbol = false;
+
+            for (int i = 0; i < length; i++)
+            {
+                int charType = rand() % 4;
+                int index;
+                switch (charType)
+                {
+                case 0: // lowercase letter
+                    hasLowercase = true;
+                    if (!(hasLowercase && hasUppercase && hasDigit && hasSymbol)){
```

Comment:
**question**: I don't understand this if statement. If the password is missing a lowercase or an uppercase or a digit or a symbol, stop the loop? What if the password doesn't have the minimum # of characters yet?

Reply:
this just goes to the next iteration of the loop if all the reqs aren't satisfied yet, so i should change it to a while loop